apps/files/src/components/FileItem.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <oc-file
+    :name="fileName"
+    :extension="item.extension"
+    :icon="previewIcon"
+    :iconUrl="previewUrl"
+    :filename="item.name"
+    :data-preview-loaded="previewLoaded"
+    />
+</template>
+<script>
+import queryString from 'query-string'
+import Mixins from '../mixins'
+
+export default {
+  name: 'FileItem',
+  mixins: [
+    Mixins
+  ],
+  props: {
+    item: {
+      type: Object
+    },
+    davUrl: {
+      type: String
+    },
+    showPath: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data: function () {
+    return {
+      previewUrl: this.item.previewUrl,
+      // false while the preview is loading
+      // true when the preview loading process is done,
+      // even if we fell back to the mime type icon
+      previewLoaded: false
+    }
+  },
+  computed: {
+    fileName () {
+      if (this.showPath) {
+        const pathSplit = this.item.path.substr(1).split('/')
+        if (pathSplit.length === 2) return `${pathSplit[pathSplit.length - 2]}/${this.item.basename}`
+        if (pathSplit.length > 2) return `…/${pathSplit[pathSplit.length - 2]}/${this.item.basename}`
+      }
+      return this.item.basename
+    },
+    previewIcon () {
+      return this.fileTypeIcon(this.item)
+    }
+  },
+  mounted () {
+    this.loadPreview()
+  },
+  methods: {
+    loadPreview () {
+      if (this.item.previewUrl) {
+        this.previewUrl = this.item.previewUrl
+        this.previewLoaded = true
+        return
+      }
+
+      // TODO: check if previews are globally enabled (requires capability entry)
+      // don't load previews for pending or rejected shares (status)
+      if (!this.davUrl || this.item.type === 'folder' || (typeof this.item.status !== 'undefined' && this.item.status !== 0)) {
+        this.previewLoaded = true
+        return
+      }
+      const query = queryString.stringify({
+        x: this.thumbDimensions,
+        y: this.thumbDimensions,
+        c: this.item.etag,
```

Comment:
I think after `stringify` the etag will have `"` around it. They should be remove if we want to be equal to the OC10 url

Reply:
Nice one!

I just first looked into stripping the double quote at the source, but realized that it would affect other code paths that would require adding back the double quotes, so decided against a global solution.

I'll then push a fix that removes the double quotes locally, also applied to the media viewer code that has a similar logic.